src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/PublicSymbols/AnonymousType.DelegatePublicSymbol.cs

```diff
@@ -28,10 +30,9 @@ internal override NamedTypeSymbol MapToImplementationSymbol()
 
             internal override AnonymousTypeOrDelegatePublicSymbol SubstituteTypes(AbstractTypeMap map)
             {
-                var typeDescr = TypeDescriptor.SubstituteTypes(map, out bool changed);
-                return changed ?
-                    new AnonymousDelegatePublicSymbol(Manager, typeDescr) :
-                    this;
+                var containingTypeParameters = map.SubstituteTypeParameters(ContainingTypeParameters);
```

Comment:
>SubstituteTypeParameters [](http://example.com/codeflow?start=51&length=24)

Usage of the API in this manner doesn't look safe. A type parameter doesn't have to map to a type parameter. It might map to some other type.